HotelReservation/HotelReservationTests/ReserveHotelUsecaseTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿using HotelReservation.Hotels;
+using NUnit.Framework;
+using FakeItEasy;
+using System.Collections.Generic;
+using HotelReservation;
+using HotelReservation.ReservationSteps;
+
+namespace HotelReservationTests
+{
+    [TestFixture]
+    class ReserveHotelUsecaseTests
+    {
+        ReserveHotelUsecase _subject;
+        HotelSystem _hotelSystemDouble;
+        StepFactory _stepFactoryDouble;
+        StepsExecutor _stepExecutorDouble;
+
+        [SetUp]
+        public void BeforeTest()
+        {
+            _hotelSystemDouble = A.Fake<HotelSystem>();
+            _stepFactoryDouble = A.Fake<StepFactory>();
+            _stepExecutorDouble = A.Fake<StepsExecutor>();
+            _subject = new ReserveHotelUsecase(_hotelSystemDouble, _stepFactoryDouble, _stepExecutorDouble);
+        }
+
+        [Test]
+        public void GetHotelsShouldCallHotelSystemGetHotels()
```

Comment:
Try to avoid writing test names based on method or classes names. This test name will become outdated if any of these names changes and you won't have any reference to it.

Instead, show the intention:
ShouldRetrieveAvailableHotels()

Reply:
Fixed